snippets/olm-psa-opm-catalog-source-restricted-known-issue.adoc

```diff
@@ -0,0 +1,25 @@
+// Text snippet included in the following assemblies:
+//
+// * 
+//
+// Text snippet included in the following modules:
+//
+// * olm-catalogsource.adoc
+// * olm-creating-catalog-from-index
+
+:_content-type: SNIPPET
+
+There is a known issue in the `opm` CLI tool that prevents SQLite-based catalog source pods from reaching a ready state under restricted pod security enforcement. (link:https://issues.redhat.com/browse/OCPBUGS-122[*OCPBUGS-122*])
+
+Pod security admission was introduced in Kubernetes 1.24 to ensure pod security standards. In {product-title} {product-version}, namespaces do not have restricted pod security enforcement by default and the default security mode is set to `legacy`.
+
+It is planned that all namespaces will default to restricted pod security standard enforcement in a future release of {product-title}. It is recommended that you take action now to ensure your catalog sources run under restricted pod security enforcement. If you do not take action to run your catalog sources under restricted enforcement, your catalog sources might not run.
+
+Catalog authors can fix the known issue by updating their catalog image with a newer version of the `opm` CLI tool or migrating their catalog to the file-based catalog format. 
```

Comment:
It might just be me, but it kind of seems like you're mixing two things in this snippet - the known issue and the future restricted enforcement. You touch on the known issue, then go to the future restricted enforcement, then back to how to fix the known issue.

I assume that even if that bug gets fixed, this part about the future restricted enforcement coming would still be applicable? If that's the case, then you might want to consider separating the two (if that makes sense to do). Unless the bug is the only reason that they'd have to take action - and if that bug gets fixed then there's no longer any action they'd need to take in 4.13?

Reply:
the "bug" (i'm reluctant to even call it a bug, it's a limitation that was irrelevant prior to PSA enforcement) is in the opm binary itself which is part of every catalog image.

But yes, the net is:

1) if you've got a catalog image containing the older opm binary, the catalog image cannot run in restricted environments, so you need to take other actions to run the catalog as privileged, and ensure the NS will allow the catalog to run privileged.
2) for now, that is in 4.11/4.12, no user action is required because 4.11/4.12 will default to running your catalog as privileged and the NS will allow catalogs to run privileged.
3) in the future (likely 4.13, but who knows) we will default to running your catalog as restricted, and the NS will default to only allowing restricted workloads.  So at that point if your catalog image can't run restricted (because you haven't updated it with a new opm binary), you will need to take action to force the catalog to run privileged, and force the NS to allow privileged workloads
4) you can take those actions(update the opm binary, or configure the catalog explicitly to run privileged and configure the NS explicitly to allow privileged workloads) now (in 4.12), even though they are not strictly necessary yet, in order to ensure you don't get broken in the future(4.13+).

Reply:
Okay thanks - so maybe calling it a "known issue" isn't really the right term, if it's not planning on being fixed. I just opened the bug now and see that it's been closed as not a bug. Might want to classify it a bit differently in this text then.

Reply:
I am going to sync with @mwerner2113 to rework this.